module/product/src/Infrastructure/Strategy/ProductStrategyInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Product\Infrastructure\Strategy;
+
+use Ergonode\Product\Domain\Entity\AbstractProduct;
+use Ergonode\Product\Domain\ValueObject\Sku;
+use Ergonode\SharedKernel\Domain\Aggregate\ProductId;
+use Ergonode\SharedKernel\Domain\Aggregate\TemplateId;
+
+interface ProductStrategyInterface
+{
+    public function supports(string $type): bool;
+
+    public function build(
+        ProductId $id,
+        Sku $sku,
+        TemplateId $templateId,
+        array $categories,
+        array $attributes
+    ): AbstractProduct;
```

Comment:
IMHO  this inteface should be open for any potential informations that be needed for catering any product Class

Reply:
Maybe a second interface can be created extending this one with an additional optional parameter.